File: generate_dark_images.js
const fs = require('fs');
const path = require('path');

const imagesDir = path.join(__dirname, 'public', 'images');
const imageNames = [
  'pos-1.png',
  'pos-2.png',
  'front-office.png',
  'housekeeping.png',
  'food-beverage.png',
  'purchasing.png',
  'accounting.png',
  'hrd.png',
  'self-order.png',
  'employee-portal.png'
];

// A tiny valid 1x1 transparent PNG base64
const dummyPngBase64 = 'iVBORw0KGgoAAAANSUhEUgAAAAEAAAABCAQAAAC1HAwCAAAAC0lEQVR42mNkYAAAAAYAAjCB0C8AAAAASUVORK5CYII=';
const dummyBuffer = Buffer.from(dummyPngBase64, 'base64');

imageNames.forEach(name => {
  const srcPath = path.join(imagesDir, name);
  const darkName = name.replace('.png', '-dark.png');
  const destPath = path.join(imagesDir, darkName);

  if (fs.existsSync(srcPath)) {
    // Copy existing light image to dark image as a placeholder
    fs.copyFileSync(srcPath, destPath);
    console.log(`Copied ${name} to ${darkName}`);
  } else {
    // Create tiny dummy png for both light and dark if they don't exist
    fs.writeFileSync(srcPath, dummyBuffer);
    fs.writeFileSync(destPath, dummyBuffer);
    console.log(`Created dummy placeholders for ${name} and ${darkName}`);
  }
});

console.log('Dummy image generation complete!');

File: generate_dark_images.py
import os
from PIL import Image, ImageOps

images_dir = r"f:\WEB-SERVER\WEB\setara-landing-marketing\public\images"
image_names = [
    "pos-1.png",
    "pos-2.png",
    "front-office.png",
    "housekeeping.png",
    "food-beverage.png",
    "purchasing.png",
    "accounting.png",
    "hrd.png",
    "self-order.png",
    "employee-portal.png"
]

print("Starting dark image generation...")

for name in image_names:
    src_path = os.path.join(images_dir, name)
    dest_path = os.path.join(images_dir, name.replace(".png", "-dark.png"))
    
    # If the source image does not exist, create a simple dark placeholder
    if not os.path.exists(src_path):
        print(f"Source {name} not found. Creating a dark placeholder...")
        img = Image.new("RGB", (800, 600), color=(20, 20, 20))
        img.save(dest_path)
        # also create a light placeholder so we don't get 404
        img_light = Image.new("RGB", (800, 600), color=(240, 240, 240))
        img_light.save(src_path)
        continue

    try:
        # Load light image
        img = Image.open(src_path).convert("RGBA")
        
        # Split into channels
        r, g, b, a = img.split()
        
        # Invert colors for RGB channels
        rgb_img = Image.merge("RGB", (r, g, b))
        inverted_rgb = ImageOps.invert(rgb_img)
        
        # Re-merge with original alpha or a slightly darkened transparent layer
        dark_img = Image.merge("RGBA", (inverted_rgb.split()[0], inverted_rgb.split()[1], inverted_rgb.split()[2], a))
        
        # Save dark version
        dark_img.save(dest_path)
        print(f"Created dark version: {dest_path}")
    except Exception as e:
        print(f"Failed to process {name}: {e}")
        # Fallback to simple copy or simple placeholder
        try:
            img = Image.new("RGB", (800, 600), color=(20, 20, 20))
            img.save(dest_path)
            print(f"Created fallback placeholder for {name}")
        except Exception:
            pass

print("Done!")
